### PR TITLE
Implement the javac_workers in the please java toolchain 

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -65,7 +65,8 @@ if ! hash gold 2>/dev/null ; then
     warn "Gold not found, excluding Gold tests"
     EXCLUDES="${EXCLUDES} --exclude=gold"
 fi
-if ! hash java 2>/dev/null ; then
+
+if [ ]! hash java 2>/dev/null ] && [ "$GOOS" != "linux" ] ; then
```
Comment (peterebden) [inline bootstrap.sh:69]: the start of this line looks wrong (extra `]`?)

File: rules/java_rules.build_defs [hunk 2]
```diff
@@ -509,56 +513,65 @@ def _jarcat_cmd(main_class=None, preamble=None, manifest=None):
...
         outs = [name],
         tools = [CONFIG.JARCAT_TOOL],
         cmd = " && ".join([
-            '"$TOOL" x $SRCS -o extract_dir',
-            'mkdir $OUT && mkdir $OUT/bin',
-            'mv $(find extract_dir -name javac) $OUT/bin',
-            'mv $(find extract_dir -name java) $OUT/bin',
-            'mv $(find extract_dir -name lib) $OUT',
+            'mkdir extract_dir',
+            'tar -xf $SRCS -C extract_dir',
```
Comment (Tatskaari) [inline rules/java_rules.build_defs:569]: Will change this back to jarcat when I merge the doco PR in

File: rules/java_rules.build_defs [hunk 3]
```diff
@@ -509,56 +513,65 @@ def _jarcat_cmd(main_class=None, preamble=None, manifest=None):
...
+                    download_hashes = []) -> str:
     """Downloads the JDK and sets up build rule metadata so the language rules can use the toolchain"""
-    current_package = package_name()
-
     jdk_url = jdk_url if isinstance(jdk_url, str) else jdk_url[CONFIG.OS]
 
     if not jdk_url:
         raise ParseError(f"No java jdk_url defined for platform {CONFIG.OS}")
 
+    this = canonicalise(f":{name}")
```
Comment (peterebden) [inline rules/java_rules.build_defs:524]: do you need to canonicalise this? it shouldn't matter?

Comment (Tatskaari) [reply rules/java_rules.build_defs:524]: Yeah we do, we're passing it via the metadata and it's used in the java_library and java_test rules.

**Pull request merged**